src/components/dialog/tests/constraints.e2e.ts

```diff
@@ -0,0 +1,110 @@
+import { E2EElement, E2EPage, newE2EPage } from "@stencil/core/testing";
+
+describe("[ch-dialog][semantic]", () => {
+  let page: E2EPage;
+  let chDialogRef: E2EElement;
+
+  beforeEach(async () => {
+    page = await newE2EPage({
+      html: `<ch-dialog></ch-dialog>`,
+      failOnConsoleError: true
+    });
+
+    chDialogRef = await page.find("ch-dialog");
+
+    // Set properties
+    chDialogRef.setProperty("showHeader", true);
+    await page.waitForChanges();
+    chDialogRef.setProperty("showFooter", true);
+    await page.waitForChanges();
+    chDialogRef.setProperty("resizable", true);
+    await page.waitForChanges();
+    chDialogRef.setProperty("show", true);
+    await page.waitForChanges();
+  });
+
+  it("should not render a header element", async () => {
+    const headerEl = await page.find("ch-dialog >>> header");
+    expect(headerEl).toBeNull();
+  });
+
+  it("should not render a footer element", async () => {
+    const footerEl = await page.find("ch-dialog >>> footer");
+    expect(footerEl).toBeNull();
+  });
+
+  it("should not include a role='banner' attribute on the part='header' element", async () => {
+    const headerPartRef = await page.find("ch-dialog >>> [part='header']");
+    await page.waitForChanges();
+    expect(headerPartRef).not.toBeNull();
+
+    const roleAttribute = headerPartRef.getAttribute("role");
+    expect(roleAttribute).not.toBe("banner");
+  });
```

Comment:
This test only checks if that the header part does not have a role. You should not check if the part exists. That belongs to another test.

```suggestion
  it("should not include a role='banner' attribute on the part='header' element", async () => {
    const headerPartRef = await page.find("ch-dialog >>> [part='header']");
    expect(headerPartRef).not.toHaveAttribute("role");
  });
```